lightningd/status.c

```diff
@@ -62,13 +65,19 @@ static void status_send_with_hdr(u16 type, const void *p, size_t len)
 
 	be_type = cpu_to_be16(type);
 	be_len = cpu_to_be16(len + sizeof(be_type));
-	assert(status_fd >= 0);
 	assert(be16_to_cpu(be_len) == len + sizeof(be_type));
 
-	if (!write_all(status_fd, &be_len, sizeof(be_len))
-	    || !write_all(status_fd, &be_type, sizeof(be_type))
-	    || !write_all(status_fd, p, len))
-		err(1, "Writing out status %u len %zu", type, len);
+	if (status_fd >= 0) {
+		if (!write_all(status_fd, &be_len, sizeof(be_len))
+		    || !write_all(status_fd, &be_type, sizeof(be_type))
+		    || !write_all(status_fd, p, len))
+			err(1, "Writing out status %u len %zu", type, len);
+	} else {
+		u8 *msg = tal_arr(NULL, u8, sizeof(be_type) + len);
```

Comment:
Is this using `memcpy` instead of the `towire_*` functionality to avoid having to include the `towire.h` header?